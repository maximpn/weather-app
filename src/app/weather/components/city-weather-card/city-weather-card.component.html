<mat-card>
  <mat-card-header>
    <div mat-card-avatar>
      <wa-weather-condition-icon [weatherCondition]="weather.weather[0]"></wa-weather-condition-icon>
    </div>
    <mat-card-title>{{ weather.name }} {{ weather.main.temp | number:'1.0-0' }}Â°C</mat-card-title>
    <mat-card-subtitle>Wind: {{ weather.wind.speed }} km/h</mat-card-subtitle>
  </mat-card-header>
</mat-card>
