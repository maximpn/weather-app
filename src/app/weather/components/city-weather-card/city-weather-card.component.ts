import { Component, Input } from '@angular/core';

import { CityWeather } from '../../models/city-weather.model';

@Component({
  selector: 'wa-city-weather-card',
  templateUrl: 'city-weather-card.component.html',
})
export class CityWeatherCardComponent {
  @Input()
  weather!: CityWeather;

  get weatherIconLink(): string {
    return `http://openweathermap.org/img/wn/${this.weather.weather[0].icon}.png`;
  }

  get weatherDescription(): string {
    return this.weather.weather[0].description;
  }
}
