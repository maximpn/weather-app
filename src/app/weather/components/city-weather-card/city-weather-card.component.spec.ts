import { Component, Input, NO_ERRORS_SCHEMA } from '@angular/core';
import { ComponentFixture, TestBed } from '@angular/core/testing';

import { CityWeather } from '../../models/city-weather.model';
import { WeatherCondition } from '../../models/weather-condition.model';

import { CityWeatherCardComponent } from './city-weather-card.component';

@Component({
  selector: 'wa-weather-condition-icon',
  template: '{{ weatherCondition | json }}',
})
class WeatherConditionIconMockComponent {
  @Input()
  weatherCondition!: WeatherCondition;
}

describe('CityWeatherCardComponent', () => {
  let fixture: ComponentFixture<CityWeatherCardComponent>;

  beforeEach(() => {
    fixture = TestBed.configureTestingModule({
      declarations: [
        CityWeatherCardComponent,
        WeatherConditionIconMockComponent,
      ],
      schemas: [
        NO_ERRORS_SCHEMA,
      ],
    }).createComponent(CityWeatherCardComponent);

    fixture.componentInstance.weather = {
      name: 'City',
      main: {
        temp: 30.55,
      },
      wind: {
        speed: 5.5,
      },
      weather: [
        { id: 1, main: 'cloudy', description: 'cloudy', icon: '02d' },
      ],
    } as CityWeather;

    fixture.detectChanges();
  });

  it('should render the component', () => {
    expect(fixture.nativeElement).toMatchSnapshot();
  });
});
