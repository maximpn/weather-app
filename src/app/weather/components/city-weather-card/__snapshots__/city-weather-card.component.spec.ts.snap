// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityWeatherCardComponent should render the component 1`] = `
<div
  id="root0"
>
  <mat-card>
    <mat-card-header>
      <div
        mat-card-avatar=""
      >
        <wa-weather-condition-icon>
          {
  "id": 1,
  "main": "cloudy",
  "description": "cloudy",
  "icon": "02d"
}
        </wa-weather-condition-icon>
      </div>
      <mat-card-title>
        City 31Â°C
      </mat-card-title>
      <mat-card-subtitle>
        Wind: 5.5 m/s
      </mat-card-subtitle>
    </mat-card-header>
  </mat-card>
</div>
`;
